pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -135,14 +135,21 @@ var _ Manager = &manager{}
 func NewManager(topology []cadvisorapi.Node, topologyPolicyName string, topologyScopeName string, topologyPolicyOptions map[string]string) (Manager, error) {
 	klog.InfoS("Creating topology manager with policy per scope", "topologyPolicyName", topologyPolicyName, "topologyScopeName", topologyScopeName)
 
-	opts, err := NewPolicyOptions(topologyPolicyOptions)
-	if err != nil {
-		return nil, err
-	}
-
-	numaInfo, err := NewNUMAInfo(topology, opts)
-	if err != nil {
-		return nil, fmt.Errorf("cannot discover NUMA topology: %w", err)
+	var (
+		opts     PolicyOptions
+		numaInfo *NUMAInfo
+		err      error
+	)
+
+	if topologyPolicyName != PolicyNone {
+		opts, err = NewPolicyOptions(topologyPolicyOptions)
+		if err != nil {
+			return nil, err
+		}
+		numaInfo, err = NewNUMAInfo(topology, opts)
+		if err != nil {
+			return nil, fmt.Errorf("cannot discover NUMA topology: %w", err)
+		}
```

Comment:
this is the most straightforward approach and probably the best start.
But entering the `switch` below with `nil` values, which is unprecedented, gives me a fuzzy feeling.
We should explore if there are ways to rethink the flow in this function such like (brainstorming now) treating `PolicyNone` as special case and short circuiting out if it is.
If turns out there's no cleaner way to handle this case, let's go with the simple approach. But we should try first.

Reply:
we can do a quick check for policyName is none, and wrap a function to getScope by policy.
this way, we can remove the rest checks for `case PolicyNone` in `NewManager`, will this be better?
```
func NewManager() {
	if topologyPolicyName == PolicyNone {
		policy := NewNonePolicy()
		scope,err := getScope(topologyScopeName,policy)
		if err!=nil {
			return nil, err
		}
		manager := &manager{
			scope: scope,
		}
		return manager, nil	
	}
       //continue the logic
}

func getScope(topologyScopeName string, policy Policy) (scope Scope, err error) {
	switch topologyScopeName {
	case containerTopologyScope:
		scope = NewContainerScope(policy)

	case podTopologyScope:
		scope = NewPodScope(policy)

	default:
		return nil, fmt.Errorf("unknown scope: \"%s\"", topologyScopeName)
	}
	return scope, nil
}

```

Reply:
I'll think about options and come back with suggestions. We totally should not wait for this fix too long, but let's also take some time to evaluate alternatives, the code is simple enough and well covered by tests so we have some room for exploration.

Reply:
We could just add new scope called `scopeNone` which would just noop and return early like this:
```go
func NewManager(paramList) (Manager, error) {
    if topologyPolicyName == PolicyNone {
        return &manager{
            scope: scopeNone(),
        }, nil
    }
    ... // continue with other policies and scopes
}
```

Reply:
Yes.

Reply:
> We could just add new scope called `scopeNone` which would just noop and return early like this:
> 
> ```go
> func NewManager(paramList) (Manager, error) {
>     if topologyPolicyName == PolicyNone {
>         return &manager{
>             scope: scopeNone(),
>         }, nil
>     }
>     ... // continue with other policies and scopes
> }
> ```
> 
> Right now both `container` and `pod` scope has this check at the beginning of `Admit` func:
> 
> ```go
> func (s *containerScope) Admit(pod *v1.Pod) lifecycle.PodAdmitResult {
> 	// Exception - Policy : none
> 	if s.policy.Name() == PolicyNone {
> 		return s.admitPolicyNone(pod)
> 	}
> ```
> 
> we could get rid of those with this approach.

I tend to like this direction, I think it has potential. Let's evaluate the full change and see how it works.
Thanks for chiming in @PiotrProkop ! Unforetunately I was up until now more busy than expected prepping for kubecon eu.